libmcount/mcount.c

```diff
@@ -909,7 +909,9 @@ static void mcount_startup(void)
 		symtabs.flags &= ~SYMTAB_FL_SKIP_DYNAMIC;
 
 	mcount_exename = read_exename();
-	record_proc_maps(dirname, session_name(), &symtabs);
+	const char *session_id = session_name();
+	record_proc_maps(dirname, session_id, &symtabs);
+	set_kaddr(dirname, session_id);
```

Comment:
I don't like putting declarations in the middle.  Actually `session_name()` is intended to be called directly - IOW it's ok to call the function multiple times.  So you can just add:

```
set_kaddr(dirname, session_name());
```